bayesian/utility/v2/component_analyses.py

```diff
@@ -114,13 +119,50 @@ def ca_validate_input(input_json: Dict, ecosystem: str) -> Tuple[List[Dict], Lis
 
         if ecosystem == 'golang':
             _, clean_version = GolangDependencyTreeGenerator.clean_version(given_version)
+            pseudo_version = is_pseudo_version(ecosystem, clean_version)
 
         packages_list.append(
-            {"name": package, "version": clean_version, 'given_version': given_version})
-        normalised_input_pkgs.append(normlize_packages(package, clean_version, given_version))
+            {"name": package, "version": clean_version, "given_version": given_version,
+             "is_pseudo_version": pseudo_version})
+        normalised_input_pkgs.append(normlize_packages(package, clean_version,
+                                                       given_version, pseudo_version))
     return packages_list, normalised_input_pkgs
 
 
+def get_batch_ca_data(ecosystem: str, packages) -> dict:
+    """Fetch package details for component analyses."""
+    logger.debug('Executing get_batch_ca_data')
+
+    aggregated_response = {}
+    semver_packages = []
+    pseudo_version_packages = []
+
+    # Need to seperate semver and pseudo verion packages for golang
+    if (ecosystem == "golang"):
+        for p in packages:
+            if p['is_pseudo_version']:
```

Comment:
Wont you get only modules from here? packages wont have that pseudo version attached. If yes, then your previous query etc based on the 'package_name' will be incorrect.

Reply:
'is_pseudo_version' is a new flag set by this code for each package object (which has package and module data). If you check 'ca_validate_input()' above this code, pseudo version flag is set for every package in the request based on version string in the request.

Reply:
@dgpatelgit Thats the point im trying to make. I want to understand the reason for fetching using the package name instead of module name. Whats the reason behind it? pseudo versions are added at the module level.

Reply:
Vulnerability node has "module_name" and "package_name", so instead of getting all vulns for module and then filtering response using package_name, I thought of getting vuln using "package_name" itself. As you mentioned in the slack during discussion, that for vuln at module level, vuln node will have module_name == pacakge_name.

Reply:
ok.I get what you are trying to do here. Instead of module you want to query package. That will work. no problems. Please cross verify the response time. As the number of packages will be more, it might take more time to get response from gremlin. So please gauge both the approaches and select which suits best.

Reply:
I have check  the performance on module_name and package_name queries, on a small sample data I noticed that module_name took around 350ms to get vuln data for modules and all its packages, on top API server need to filter required packages from response. Whereas second query based on package_name took 250ms and completely avoided response filtering in API server. So with this data, I took a call to fetch gremlin vuln data based on package_name filter.